examples/gno.land/p/demo/diff/diff.gno

```diff
@@ -0,0 +1,151 @@
+package diff
+
+import (
+	"strings"
+)
+
+type EditType int
```

Comment:
You're right. Using `byte` instead of `rune` breaks the support of multi-byte UTF-8 characters. The delta between old and new would still be correct, as long as we don't try to print it, but it is not the use case we want. Your approach is definitely correct. 

By the way, your implementation is better that gnu-diff which is flawed in the same way: https://lists.gnu.org/archive/html/bug-diffutils/2018-04/msg00009.html. I suspect that other implementations (if not most) are incorrect too.

Could you add a simple test involving multi-byte characters to demonstrate that?